tests/unittests/MatrixRotationTest.cpp

```diff
@@ -0,0 +1,676 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file tests a toy neural network that can recognize if a matrix 3x3 is
+// the rotation of another matrix 3x3.
+// This is *not* about rotation matrices used for computer graphics, but a much
+// simpler concept.
+// Informally, let M1 and M2 be two 3x3 matrices, M2 is a rotation of M1 if it
+// exists a way to rotate the cells of M1 to get M2.
+// E.g.,
+// Rotate clockwise 1 cell:
+//              --+
+//      | a b c | |
+// M1 = | d e f | V
+//      | g h i |
+// =>
+//      | d a b |
+// M2 = | g e c |
+//      | h i f |
+
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+#include "glow/Graph/Graph.h"
+#include "glow/IR/IR.h"
+#include "glow/IR/IRBuilder.h"
+#include "glow/IR/Instrs.h"
+#include "glow/Support/Random.h"
+
+#include "gtest/gtest.h"
+
+#include "llvm/ADT/Twine.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <cstddef> // for size_t.
+#include <cstdint> // for uint8_t.
+#include <utility> // for std::pair.
+
+using namespace glow;
+
+namespace {
+template <typename Element> struct Matrix3x3 {
+  Element m_[3][3];
```

Comment:
maybe just std::array of std::array.  ctor and operator== will be trivial in that case.